v2/internal/testcommon/vcr/v3/test_recorder.go

```diff
@@ -91,22 +89,25 @@ func NewTestRecorder(
 			return false
 		}
 
-		// verify custom request count header (see counting_roundtripper.go)
-		if r.Header.Get(vcr.COUNT_HEADER) != i.Headers.Get(vcr.COUNT_HEADER) {
-			return false
+		// verify custom request count header matches, if present
+		if header := r.Header.Get(COUNT_HEADER); header != "" {
+			if header != i.Headers.Get(COUNT_HEADER) {
+				return false
+			}
 		}
 
-		if r.Body == nil {
-			return i.Body == ""
+		// verify custom body hash header matches, if present
+		if header := r.Header.Get(HASH_HEADER); header != "" {
+			if header != i.Headers.Get(HASH_HEADER) {
+				return false
+			}
 		}
 
-		var b bytes.Buffer
-		if _, err := b.ReadFrom(r.Body); err != nil {
-			panic(err)
+		if r.Body == nil {
+			return i.Body == ""
```

Comment:
Is this just a quirk of vcr that empty bodies are empty strings?

_Maybe_ worth a comment?

Reply:
Yes. Added a comment.